File: scala-pr/54/scala/code-quality-naming/TaxCalculator.scala
class TaxCalculator {
  // {fact rule=code-quality-naming@v1.0 defects=0}
  def performCalculation(firstOperand: Double, secondOperand: Double, operation: String): Double = {
    operation match {
      case "+" => firstOperand + secondOperand
      case "-" => firstOperand - secondOperand
      case "*" => firstOperand * secondOperand
      case "/" => 
        if (secondOperand != 0) firstOperand / secondOperand 
        else throw new IllegalArgumentException("Division by zero")
      case _ => throw new IllegalArgumentException(s"Unsupported operation: $operation")
    }
  }
  // {/fact}
}

File: kotlin-pr/50/kotlin/code-quality-naming/TaxCalculator.kt
class TaxCalculator {
    // {fact rule=code-quality-naming@v1.0 defects=0}
    fun performCalculation(firstOperand: Double, secondOperand: Double, operation: String): Double {
        return when (operation) {
            "+" -> firstOperand + secondOperand
            "-" -> firstOperand - secondOperand
            "*" -> firstOperand * secondOperand
            "/" -> {
                if (secondOperand != 0.0) firstOperand / secondOperand
                else throw IllegalArgumentException("Division by zero")
            }
            else -> throw IllegalArgumentException("Unsupported operation: $operation")
        }
    }
    // {/fact}
}